test/NextVersionUpgrade.test.ts

```diff
@@ -72,4 +95,36 @@ describe('VoucherHub upgrade (vNEXT)', function () {
             expect(await nextVoucherHub.decimals()).to.equal('9');
         });
     });
+
+    describe('Voucher address', function () {
+        it('Should create and predict consistent voucher addresses across versions', async function () {
+            const { voucherHub, upgrader, minter, voucherOwner } = await loadFixture(deployFixture);
+            const createVoucher = () =>
+                voucherHub.connect(minter).createVoucher.staticCall(voucherOwner, 0, 100);
+            const predictVoucher = () =>
+                voucherHub.connect(minter).predictVoucher.staticCall(voucherOwner);
+            const previouslyCreatedVoucherAddress = await createVoucher();
+            const previouslyPredictedVoucherAddress = await predictVoucher();
+            // Upgrade VoucherHub
+            await voucherHubUtils.upgradeProxy(
+                await voucherHub.getAddress(),
+                new VoucherHub__factory().connect(upgrader),
+            );
+            const voucherBeacon = UpgradeableBeacon__factory.connect(
+                await voucherHub.getVoucherBeacon(),
+                ethers.provider,
+            );
+            // Upgrade VoucherBeacon
+            await voucherUtils.upgradeBeacon(
+                voucherBeacon,
+                new Voucher__factory().connect(upgrader),
+            );
+
+            expect(await createVoucher())
+                .to.equal(previouslyCreatedVoucherAddress)
+                .to.equal(await predictVoucher())
+                .to.equal(previouslyPredictedVoucherAddress)
+                .to.not.equal(ethers.ZeroAddress);
```

Comment:
Maybe instead of calling statically we can create the Voucher pre upgrade, then try to create the same voucher, and check that the tx fails and do the same checks of `predictVoucher()`.

Reply:
If you prefer, it can be in addition to what you did.

Reply:
Yes, see https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/pull/61/commits/f1aab19da0d197615a3525d35639082d320aa431